utils/utils.php

```diff
@@ -618,13 +618,8 @@ function tlms_recordLog($message){
 if(!function_exists('tlms_passgen')){
     function tlms_passgen($length = 8){
 
-        $characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
-        $randomString = '';
-        for ($i = 0; $i < $length; $i++) {
-            $randomString .= $characters[rand(0, strlen($characters) - 1)];
-        }
-
-        return $randomString;
+        $length = max($length, 8);
+        return wp_generate_password($length) . 'aA1'; // aA1 ensures that password has one uppercase letter, one lowercase and one digit
```

Comment:
You can make aA1 random by creating 3 pools of characters
a-z, A-Z, 0-9 and adding a single character from each pool.

At the end you can pass the string from str_shuffle() to make sure that those characters are not always positioned at the end of the string

Reply:
[4b5578b](https://github.com/epignosis/talentlms-wordpress/pull/6/commits/4b5578b3b6fdead109e926493799f2840b5711e9)